src/BoundaryConditions/fill_halo_regions_open.jl

```diff
@@ -1,39 +1,107 @@
-#####
-##### Outer functions for filling halo regions for open boundary conditions.
-#####
-
-# TODO: support true open boundary conditions.
-# For this we need to have separate functions for each of the six boundaries,
-# and need to unroll a loop over the boundary normal direction.
-# The syntax for `getbc` is also different for OpenBoundaryCondition than for others,
-# because the boundary-normal index can vary (and array boundary conditions need to be
-# 3D in general).
-
-@kernel function set_west_or_east_u!(u, i_boundary, bc, grid, args) 
+@inline fill_open_boundary_regions!(field, args...) = 
+    fill_open_boundary_regions!(field, field.boundary_conditions, field.indices, instantiated_location(field), field.grid)
+
+"""
+    fill_open_boundary_regions!(fields, boundary_conditions, indices, loc, grid, args...; kwargs...)
+
+Fill open boundary halo regions by filling boundary conditions on field faces with `open_fill`. 
+"""
+function fill_open_boundary_regions!(field, boundary_conditions, indices, loc, grid, args...; kwargs...)
+    arch = architecture(grid)
+
+    left_bc = left_velocity_open_boundary_condition(boundary_conditions, loc)
+    right_bc = right_velocity_open_boundary_condition(boundary_conditions, loc)
+
+    open_fill, normal_fill = fill_open_halo(loc) 
+    fill_size = fill_halo_size(field, normal_fill, indices, boundary_conditions, loc, grid)
```

Comment:
I can't understand this code by reading it. What are `open_fill` and `normal_fill`? I think we either need comments or to use more interpretable names

Reply:
maybe instead of `fill_open_halo` you want to use `get_open_halo_filling_functions` or something.